userspace/otpilot/src/main.rs

```diff
@@ -163,6 +163,8 @@ fn run() -> TockResult<()> {
                             // Ignore error from writeln. There's nothing we can do here anyway.
                             let _ = writeln!(console, "Device: Error ending transaction.");
                         }
+                    } else {
+                        spi_device::get().end_transaction();
```

Comment:
Should `spi_device::get()` be pulled out into a local variable before line 155?  Will that simplify the code a bit?

Reply:
`BUSY` is not set here, so it doesn't need to be cleared.  It looks like this end-transaction path ignores `WRITE_ENABLE`.  Is it a problem that this path leaves `WRITE_ENABLE` set?  (I have no idea.)

Reply:
> Should `spi_device::get()` be pulled out into a local variable before line 155? Will that simplify the code a bit?

`spi_device` is a static object, so I decided to always explicitly call the getter to don't give a false impression about it being "just an instance".

Reply:
> `BUSY` is not set here, so it doesn't need to be cleared. It looks like this end-transaction path ignores `WRITE_ENABLE`. Is it a problem that this path leaves `WRITE_ENABLE` set? (I have no idea.)

We HAVE to clear `BUSY` here to allow the next transaction from coming in. However, the semantics for `WRITE_ENABLE` cannot be determined at this point (since an error occurred), so we leave it alone in both cases.